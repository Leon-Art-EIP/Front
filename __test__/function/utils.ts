import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
