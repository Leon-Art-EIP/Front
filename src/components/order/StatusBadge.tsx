import React from "react";
import { cn } from "../../tools/cn";

interface IStatusBadgeProps {
  text: string;
  count: number;
  isOpen: boolean;
}

const StatusBadge: React.FC<IStatusBadgeProps> = ({ text, count, isOpen }) => {
  return (
    <div
      className={cn(
        "inline-flex py-0.5 px-5 text-center justify-center text-xs rounded-2xl border-none gap-2 items-center",
        isOpen ? "bg-green-500 text-white" : "bg-red-500 text-white"
      )}
    >
      {text} ({count})
    </div>
  );
};

export default StatusBadge;
