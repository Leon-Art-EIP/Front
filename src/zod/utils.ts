export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
