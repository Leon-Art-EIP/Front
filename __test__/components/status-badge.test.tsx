import { test } from "vitest";
import StatusBadge from "../../src/components/order/StatusBadge";
import { testSnapshot } from "../function/utils";

test("StatusBadge snapshot - open state", () => {
  testSnapshot(<StatusBadge text="Open" count={5} isOpen={true} />);
});

test("StatusBadge snapshot - closed state", () => {
  testSnapshot(<StatusBadge text="Closed" count={10} isOpen={false} />);
});
