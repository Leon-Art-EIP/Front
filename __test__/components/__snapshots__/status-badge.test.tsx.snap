// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StatusBadge snapshot - closed state 1`] = `
<div
  className="inline-flex py-0.5 px-5 text-center justify-center text-xs rounded-2xl border-none gap-2 items-center bg-red-500 text-white"
>
  Closed
   (
  10
  )
</div>
`;

exports[`StatusBadge snapshot - open state 1`] = `
<div
  className="inline-flex py-0.5 px-5 text-center justify-center text-xs rounded-2xl border-none gap-2 items-center bg-green-500 text-white"
>
  Open
   (
  5
  )
</div>
`;
